longhaultests/src/iothub_longhaul.ts

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env node
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+import * as uuid from 'uuid';
+import * as async from 'async';
+import * as dbg from 'debug';
+const debug = dbg('longhaul:main');
+
+import { Registry, ConnectionString as HubConnectionString } from 'azure-iothub';
+import { ConnectionString as DeviceConnectionString } from 'azure-iot-device';
+
+import { D2CSender } from './d2c_sender';
+
+const MAX_EXECUTION_TIME = parseInt(process.env.MAX_EXECUTION_TIME_SECONDS) * 1000;
```

Comment:
>MAX_EXECUTION_TIME_SECONDS [](start = 48, length = 26)

Not sure about adding these env variables and the ones below.  I assume you're going to run these tests in a container with some container-creation script hat we haven't seen yet, and that's cool, but I'm wondering if some sort of config.json file might be a better place to hold these?  Like maybe launch the longhaul test and pass a config.json filename on the command line.  That way your execution parameters are decoupled from container creation.

Reply:
where (containers or not) it runs is the CI environment responsibility.
Regarding configuration, the reason I went with env variables is because i want to be able to set it up from the build definition page. I'll look if we can support something JSON later, for now I believe env variable are the way to go for build parameters.

---
In reply to: [177506818](https://github.com/Azure/azure-iot-sdk-node/pull/257#discussion_r177506818) [](ancestors = 177506818)